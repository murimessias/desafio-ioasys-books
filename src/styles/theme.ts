export default {
  border: {
    radius: {
      md: '0.4rem',
      rounded: '9999px',
    },
  },

  colors: {
    white: '#FFFFFF',
    caramel: '#F5F5EE',
    black: '#333333',
    gray: '#999999',
    accent: '#AB2680',
  },

  font: {
    family:
      "Heebo, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",

    sizes: {
      sm: '1.2rem',
      md: '1.4rem',
      lg: '1.6rem',
      xl: '2.8rem',
    },

    weight: {
      normal: 400,
      bold: 500,
    },
  },

  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50,
  },

  leading: {
    none: 1,
    tight: 1.25,
    normal: 1.5,
  },

  spacings: {
    xs: '1.6rem',
    sm: '2.4rem',
    md: '3.2rem',
    lg: '4.0rem',
    xl: '4.8rem',
  },
} as const
