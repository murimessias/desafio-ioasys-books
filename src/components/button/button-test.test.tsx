import { screen } from '@testing-library/react'
import { renderWithTheme } from 'resources/utils/tests/helpers'
import theme from 'styles/theme'

import { Button } from '.'

describe('<Button />', () => {
  it('should render a button', () => {
    const { container } = renderWithTheme(<Button>Entrar</Button>)

    const color = theme.colors.accent
    const fontSize = theme.font.sizes.lg

    expect(screen.getByRole('button', { name: /entrar/i })).toHaveStyle({
      color: color,
      height: '3.6rem',
      fontSize: fontSize,
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
