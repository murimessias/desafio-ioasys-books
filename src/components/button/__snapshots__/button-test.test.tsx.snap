// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a button 1`] = `
.c0 {
  background-color: #FFFFFF;
  border: none;
  border-radius: 9999px;
  color: #AB2680;
  cursor: pointer;
  font-size: 1.6rem;
  font-weight: 500;
  height: 3.6rem;
  min-width: 8rem;
  padding: 0 2.4rem;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
}

.c0:hover {
  background-color: #F5F5EE;
}

<button
  class="c0"
>
  <span>
    Entrar
  </span>
</button>
`
